.github/actions/ssh-debug/setup-ssh.sh

```diff
@@ -0,0 +1,140 @@
+#!/bin/bash -e
+
+get_authorized_keys() {
+    if [ -z "$AUTHORIZED_USERS" ] || ! echo "$AUTHORIZED_USERS" | grep -q "\b$GITHUB_ACTOR\b"; then
+        return 1
+    fi
+
+    api_response=$(curl -s "https://api.github.com/users/$GITHUB_ACTOR/keys")
+
+    if echo "$api_response" | jq -e 'type == "object" and has("message")' >/dev/null; then
+        error_msg=$(echo "$api_response" | jq -r '.message')
+        echo "Error: $error_msg"
+        return 1
+    else
+        echo "$api_response" | jq -r '.[].key'
+    fi
+}
```

Comment:
I think the secret should actually contain the SSH keys. And those should be fetched from the infra repo.

We can in terraform use a github_secret resource and folks should have to hardcode the SSH key they want to use to connect there. That avoids the API call here and avoids any risk around `GITHUB_ACTOR` somehow being an injection vector.